src/main/scala/cognite/spark/v1/TimeSeriesRelation.scala

```diff
@@ -55,29 +73,12 @@ class TimeSeriesRelation(config: RelationConfig, useLegacyName: Boolean)(val sql
       }
     }
 
-    client.timeSeries
-      .create(timeSeriesSeq)
-      .handleErrorWith {
-        case e: CdpApiException =>
-          if (e.code == 409) {
-            val legacyNameConflicts = e.duplicated.get.flatMap(j => j("legacyName")).map(_.asString.get)
-            if (legacyNameConflicts.nonEmpty) {
-              throw new IllegalArgumentException(
-                "Found legacyName conflicts, upserts are not supported with legacyName." +
-                  s" Conflicting legacyNames: ${legacyNameConflicts.mkString(", ")}")
-            }
-            val existingExternalIds =
-              e.duplicated.get.map(j => j("externalId").get.asString.get)
-            resolveConflict(existingExternalIds, timeSeriesSeq)
-          } else { IO.raiseError(e) }
-      } *> IO.unit
-  }
-
-  def resolveConflict(existingExternalIds: Seq[String], timeSeriesSeq: Seq[TimeSeriesCreate]): IO[Unit] =
-    upsertAfterConflict[TimeSeries, TimeSeriesUpdate, TimeSeriesCreate, TimeSeriesResource[IO]](
-      existingExternalIds,
+    createOrUpdateByExternalId[TimeSeries, TimeSeriesUpdate, TimeSeriesCreate, TimeSeriesResource[IO]](
+      Seq.empty,
       timeSeriesSeq,
-      client.timeSeries)
+      client.timeSeries,
+      doUpsert = doUpsert)
+  }
```

Comment:
Seems like we're no longer handling legacyName conflicts here, is that intentional?